### PR TITLE
Add meeting_user.set_data and use it in user actions.

### PR DESCRIPTION
**Pull request by reiterl**
Add tests. MeetingUser.set_data is an internal action.

### CODE REVIEW

File: openslides_backend/action/actions/meeting_user/set_data.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class MeetingUserSetData(UpdateAction):
+    """
+    Action to create, update or delete a meeting_user.
+    """
+
+    model = MeetingUser()
+    schema = DefaultSchema(MeetingUser()).get_create_schema(
+        required_properties=["meeting_id", "user_id"],
+        optional_properties=[
+            "comment",
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:22]: Mmh, I don't like it.

File: openslides_backend/action/actions/user/user_mixin.py [hunk 2]
```diff
@@ -98,3 +119,17 @@ def check_meeting_and_users(
             self.datastore.apply_changed_model(
                 user_fqid, {"meeting_id": instance.get("meeting_id")}
             )
+
+    def transfer_fields(self, instance: Dict[str, Any]) -> None:
+        meeting_user_data = {}
+        meeting_id = instance.pop("meeting_id", None)
+        for field in self.transfer_field_list:
+            if field in instance:
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/user/user_mixin.py:127]: Yes, I could, but this you it is more clear for me.

File: openslides_backend/action/actions/user/user_mixin.py [hunk 3]
```diff
@@ -98,3 +119,17 @@ def check_meeting_and_users(
             self.datastore.apply_changed_model(
                 user_fqid, {"meeting_id": instance.get("meeting_id")}
             )
+
+    def transfer_fields(self, instance: Dict[str, Any]) -> None:
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/user/user_mixin.py:123]: Rename.

File: openslides_backend/action/actions/meeting_user/set_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:22]: As this internal action should be used only in this context, it is not needed.

File: openslides_backend/action/actions/user/user_mixin.py [hunk 4]
```diff
@@ -54,6 +56,24 @@ def check_limit_of_user(self, number: int) -> None:
 
 
 class UserMixin(CheckForArchivedMeetingMixin):
+    transfer_field_list = {
+        "comment": {"type": "string"},
+        "number": {"type": "string"},
+        "structure_level": {"type": "string"},
+        "about_me": {"type": "string"},
+        "vote_weight": decimal_schema,
+        "personal_note_ids": id_list_schema,
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/user/user_mixin.py:65]: Okay. I reduce the transfer_field_list.

File: openslides_backend/action/actions/meeting_user/set_data.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            "number",
+            "structure_level",
+            "about_me",
+            "vote_weight",
+            "personal_note_ids",
+            "speaker_ids",
+            "supported_motion_ids",
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:29]: Okay, I will remove the fields here too.

File: openslides_backend/action/actions/meeting_user/set_data.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        meeting_users = self.datastore.filter(
+            "meeting_user",
+            And(
+                FilterOperator("meeting_id", "=", meeting_id),
+                FilterOperator("user_id", "=", user_id),
+            ),
+            ["id"],
+        ).values()
+        if not meeting_users:
+            res = self.execute_other_action(
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:52]: Is the way I'm handling this problem not useful? I think it is clear, what I'm doing and tests show that it works.

File: openslides_backend/action/actions/meeting_user/set_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (r-peschke) [reply openslides_backend/action/actions/meeting_user/set_data.py:22]: The idea is, to have only 1 action for meeting_user and that is this one.
Lets talk tomorrow:-)

[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:22]: I talked with  @r-peschke . We will include logic for giving the `id`, too.

File: openslides_backend/action/actions/meeting_user/set_data.py [hunk 7]
```diff
@@ -30,8 +32,8 @@ class MeetingUserSetData(UpdateAction):
     )
 
     def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (r-peschke) [inline openslides_backend/action/actions/meeting_user/set_data.py:34]: My approach:
```
meeting_id = instance.pop("meeting_id", None)
user_id = instance.pop("user_id", None)
if instance.get("id"):
   fqid = ....
   meeting_user = self.datastore.get(fqid, [], raise_exception=True)
   if meeting_id:
       assert meeting_id == meeting_user["meeting_id"], "Not permitted to change ..."
   if user_id:
       assert user_id == meeting_user["user_id"], "Not permitted to change ..."
elif meeting_id and user_id:
        meeting_users = self.datastore.filter(
            "meeting_user",
            And(
                FilterOperator("meeting_id", "=", meeting_id),
                FilterOperator("user_id", "=", user_id),
            ),
            ["id"],
        ).values()
        if not meeting_users:
            ... continue with existing code
else:
   raise exception either id nor meeting/user_ids
```

**Review by r-peschke — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/meeting_user/set_data.py (hunk 7 above)
Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:34]: It looks to complicated. I prefer my solution.

Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:34]: I am working on it. I will fill the blanks.

Comment (reiterl) [reply openslides_backend/action/actions/meeting_user/set_data.py:34]: The check_archived_meeting code is called before the update_instance code. So I need to throw the exception in the get_meeting_id method. So I overwrite it.

**Review by r-peschke — APPROVED** [review]

**Pull request merged**